src/coreclr/nativeaot/Runtime/StackFrameIterator.cpp

```diff
@@ -616,6 +617,29 @@ void StackFrameIterator::InternalInit(Thread * pThreadToWalk, NATIVE_CONTEXT* pC
 #endif // TARGET_ARM
 
 #undef PTR_TO_REG
+
+    // This function guarantees that the final initialized context will refer to a managed
+    // frame.  In the rare case where the PC does not refer to managed code (and refers to an
+    // assembly thunk instead), unwind through the thunk sequence to find the nearest managed
+    // frame.
+    // NOTE: When thunks are present, the thunk sequence may report a conservative GC reporting
+    // lower bound that must be applied when processing the managed frame.
+
+    ReturnAddressCategory category = CategorizeUnadjustedReturnAddress(m_ControlPC);
```

Comment:
I am not sure how this is related to the CET stuff

Reply:
InternalInit is supposed to leave the iterator in managed code. Until this change the InternalInit that takes `NATIVE_CONTEXT` was always starting from a location that is already in managed code.  (unlike the one that takes `PInvokeTransitionFrame`)

Now, when we receive hijack exception, we pop to the caller and perform suspension in the caller's context. 
Typically managed method that is not a reverse PInvoke must only be called by another managed method, so that would require no changes here. There are rare cases though in NativeAOT when ordinary managed methods are called from asm thunks. So it becomes possible for this code to see a location in a thunk. That is handled in the same way as in the other InternalInit method. (i.e. via `UnwindNonEHThunkSequence`)

Perhaps I should unify this code into a helper to not duplicate it in both InternalInit methods.

Reply:
I've unified the adjustment for asm thunks